src/wkwebkit.cs

```diff
@@ -61,6 +61,76 @@ interface WKBackForwardList {
 		WKBackForwardListItem ItemAtIndex (nint index);
 	}
 
+	[Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface WKContentRuleList
+	{
+		[Export ("identifier")]
+		string Identifier { get; }
+	}
+
+	[Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface WKContentRuleListStore
+	{
+		[Static]
+		[Export ("defaultStore")]
+		WKContentRuleListStore DefaultStore ();
+	
+		[Static]
+		[Export ("storeWithURL:")]
+		WKContentRuleListStore FromUrl (NSUrl url);
+	
+		[Export ("compileContentRuleListForIdentifier:encodedContentRuleList:completionHandler:")]
+		[Async]
+		void CompileContentRuleList (string identifier, string encodedContentRuleList, Action<WKContentRuleList, NSError> completionHandler);
+	
+		[Export ("lookUpContentRuleListForIdentifier:completionHandler:")]
+		[Async]
+		void LookUpContentRuleList (string identifier, Action<WKContentRuleList, NSError> completionHandler);
+	
+		[Export ("removeContentRuleListForIdentifier:completionHandler:")]
+		[Async]
+		void RemoveContentRuleList (string identifier, Action<NSError> completionHandler);
+	
+		[Export ("getAvailableContentRuleListIdentifiers:")]
+		[Async]
+		void GetAvailableContentRuleListIdentifiers (Action<NSString []> callback);
+	}
+	
+	[Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface WKHttpCookieStore
+	{
+		[Export ("getAllCookies:")]
+		[Async]
+		void GetAllCookies (Action<NSHttpCookie []> completionHandler);
+	
+		[Export ("setCookie:completionHandler:")]
+		[Async]
+		void SetCookie (NSHttpCookie cookie, [NullAllowed] Action completionHandler);
+	
+		[Export ("deleteCookie:completionHandler:")]
+		[Async]
+		void DeleteCookie (NSHttpCookie cookie, [NullAllowed] Action completionHandler);
+	
+		[Export ("addObserver:")]
+		void AddObserver (WKHttpCookieStoreObserver observer);
+	
+		[Export ("removeObserver:")]
+		void RemoveObserver (WKHttpCookieStoreObserver observer);
+	}
+
+	[Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```

Comment:
No BaseType for protocols (only Model needs one)
Normal inheritance, if needed, can be used to extend another protocol